src/autorun.ts

```diff
@@ -53,25 +86,40 @@ export const run = <T>(fn: Cb<T>): Observable<T> => new Observable(observer => {
         return false;
     }
 
+    const removeUnusedDeps = (ofType: TrackType) => {
+        for (let [key, { used, subscription, trackType }] of deps.entries()) {
+            if (used || trackType > ofType) {
+                continue;
+            }
+            subscription?.unsubscribe();
+            deps.delete(key);
+        }
+    }
+
     const runFn = () => {
         if (hasError) {
             return throwError(error);
         }
-        const prev$ = context.$;
-        const prev_ = context._;
-        context.$ = $;
-        context._ = _;
+        // Mark all deps as untracked and unused
+        for (let dep of deps.values()) {
+            dep.track = false;
+            dep.used = false;
+        }
+        const prevCtxt = {...context};
+        Object.assign(context, {$, _});
         try {
-            return of(fn());
+            const rsp = fn();
+            removeUnusedDeps(TrackType.Normal);
+            return of(rsp);
         } catch (e) {
+            removeUnusedDeps(TrackType.Weak);
```

Comment:
Smart as a fox you are, Johan!

Unlike I supposed, this works w/o a temp state! 👍
And even better: presetting everything as "untracked" actually reduces unnecessary calls! GJ!

Reply:
Thnx! :)